django/contrib/admindocs/utils.py

```diff
@@ -239,3 +240,32 @@ def remove_non_capturing_groups(pattern):
         final_pattern += pattern[prev_end:start]
         prev_end = end
     return final_pattern + pattern[prev_end:]
+
+
+@functools.lru_cache(maxsize=None)
+def _callback_strs():
+    return set()
```

Comment:
This seems like an odd construct. Perhaps we should use `threading.local` instead?

Reply:
That was a solid suggestion. Thanks, didn't know about `threading.local` or that I could use it here.